code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,401 @@
+				LAZYREMOVE(inserted_mobs, L)
+	for(var/obj/item/I in contents)
+		if(!LAZYIN(inserted_items, I))
+			LAZYADD(inserted_items, I)
+			RegisterSignal(I, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	for(var/mob/living/L in contents)
+		if(!LAZYIN(inserted_mobs, L))
+			LAZYADD(inserted_mobs, L)
+			RegisterSignal(L, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	calculate_tub_capacity()
+	update_washing_state()
+
+/// Moves all Items/Mobs out of the laundry machine onto the Washing Machine's Loc, unregisters signals, then nulls the tracking lists
+/obj/machinery/washing_machine/proc/eject_tub_contents()
+	if(bloody_mess)
+		playsound(loc, 'sound/effects/splat.ogg', 50, TRUE)
+	check_tub_contents() // one last time, we wanna make sure nothing got teleported out
+	if(LAZYLEN(inserted_items))
+		for(var/obj/item/I in inserted_items)
+			if(bloody_mess)
+				I.add_blood() // what a mess!
+			I.forceMove(loc)
+			UnregisterSignal(I, COMSIG_PARENT_QDELETING)
+		LAZYNULL(inserted_items)
+	if(LAZYLEN(inserted_mobs))
+		for(var/mob/living/L in inserted_mobs)
+			L.forceMove(loc)
+			UnregisterSignal(L, COMSIG_PARENT_QDELETING)
+		LAZYNULL(inserted_mobs)
+	calculate_tub_capacity()
+	update_washing_state()
 
-		if(length(contents) < 5)
-			if(state in list(OPEN_EMPTY, OPEN_FULL))
-				user.drop_item()
-				W.forceMove(src)
-				state = OPEN_FULL
-			else
-				to_chat(user, "<span class='warning'>The door is closed!</span>")
-		else
-			to_chat(user, "<span class='warning'>[src] is full!</span>")
-		update_icon(UPDATE_ICON_STATE)
+/// Calculates the total capacity of the contents of the tub,
+/obj/machinery/washing_machine/proc/calculate_tub_capacity()
+	var/new_capacity = 0
+	if(LAZYLEN(inserted_items))
+		for(var/obj/item/I in inserted_items)
+			new_capacity += I.w_class
+	if(LAZYLEN(inserted_mobs))
+		for(var/mob/living/L in inserted_mobs)
+			new_capacity += L.mob_size * 2
+	current_tub_capacity = new_capacity
+
+/obj/machinery/washing_machine/proc/toggle_door()
+	door_open = !door_open
+	if(door_open)
+		eject_tub_contents()
+		update_washing_state()
+		return
+	update_washing_state()
+
+/// Attempts to locate stamps or crayons for dyeing purposes
+/obj/machinery/washing_machine/proc/locate_color_source()
+	var/list/color_sources = list()
+	for(var/obj/item/toy/crayon/C in contents)
+		color_sources += C
+	for(var/obj/item/stamp/S in contents)
+		color_sources += S
+	if(!length(color_sources))
+		return
+	color_source = pick(color_sources)
+	if(istype(color_source, /obj/item/toy/crayon))
+		var/obj/item/toy/crayon/C = color_source
+		wash_color = C.dye_color
+	if(istype(color_source, /obj/item/stamp))
+		var/obj/item/stamp/S = color_source
+		wash_color = S.dye_color
+
+/obj/machinery/washing_machine/proc/start_wash_cycle()
+	if(stat & (NOPOWER|BROKEN))
+		return
+	locate_color_source()
+	washing = TRUE
+	playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+	atom_say("Beginning Wash Cycle!")
```

Comment:
Can you mention in the PR/changelog that the washing machine now speaks? Feels a bit silly to leave it out

Reply:
done